From 5653fd9fddcaedb848043d1b7d9b6ef36373141a Mon Sep 17 00:00:00 2001
From: arunkumarrajappa <arunkumarx.rajappa@intel.com>
Date: Wed, 1 Mar 2023 09:28:07 +0000
Subject: [PATCH] CTL-Fix for missing otp_svn after full flash.

Issue fix:
While on the first boot of full flash, otp_svn variable was
not updated to environment list, as drv->load() was returing Negative value.
As adding variable to environment list does not really depend on drv->load()
return status, have moved it outside of condition check.

Tested :
1. Flashed the image and system booting fine.
2. On first full flash otp_svn is updated in env list
3. Tested with kernel-A and Kernel-B Corrupt usecases, they work
as expected.

Signed-off-by: arunkumarrajappa <arunkumarx.rajappa@intel.com>
Upstream-Status: Pending
---
 env/env.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/env/env.c b/env/env.c
index 606b60bd02..290806a43d 100644
--- a/env/env.c
+++ b/env/env.c
@@ -225,12 +225,12 @@ int env_load(void)
 		 * one message.
 		 */
 		ret = drv->load();
+		/* Handle kernel FIT image switch */
+		env_set_bootcmd();
+		/*Update OTP Manifest id to env  */
+		env_set_otp_svn();
 		if (!ret) {
 			printf("OK\n");
-                        /* Handle kernel FIT image switch */
-                        env_set_bootcmd();
-                        /*Update OTP Manifest id to env  */
-                        env_set_otp_svn();
 			return 0;
 		} else if (ret == -ENOMSG) {
 			/* Handle "bad CRC" case */
-- 
2.17.1

